Automattic/pocket-casts-android
# Option dialog configuration fix

## Description

This PR fixes optionDialog configuration bug by saving and retrieving the data as a Parcelable object. 
Fixes # 
https://github.com/Automattic/pocket-casts-android/issues/1220

## Testing Instructions
1. Go to Profile
2. Go to settings(Top Right)
3. Tap Auto add to up next
4. Click Auto add limit to trigger the option Dialog
5. Change Screen Orientation
6. Observe the dialog doesn't crash

## Screenshots or Screencast 
https://github.com/Automattic/pocket-casts-android/assets/95022986/c6396a2f-88bd-4a5c-a5e1-dc5fc3c464e2


## Checklist
 
#### I have tested any UI changes...
